test/functional/test_framework/wallet.py

```diff
@@ -32,14 +32,39 @@ def __init__(self, test_node):
         self._address = ADDRESS_BCRT1_P2WSH_OP_TRUE
         self._scriptPubKey = hex_str_to_bytes(self._test_node.validateaddress(self._address)['scriptPubKey'])
 
+    def num_utxos(self):
+        return len(self._utxos)
+
     def generate(self, num_blocks):
         """Generate blocks with coinbase outputs to the internal address, and append the outputs to the internal list"""
         blocks = self._test_node.generatetoaddress(num_blocks, self._address)
         for b in blocks:
             cb_tx = self._test_node.getblock(blockhash=b, verbosity=2)['tx'][0]
-            self._utxos.append({'txid': cb_tx['txid'], 'vout': 0, 'value': cb_tx['vout'][0]['value']})
+            self._utxos.append({'txid': cb_tx['txid'], 'vout': 0, 'value': cb_tx['vout'][0]['value'], 'blockhash':b})
         return blocks
 
+    def generate_random_outputs(self, utxo, num_outputs, confirmed=True):
```

Comment:
It's true that I only use this function in `feature_bip68_sequence.py` to create confirmed outputs, but `generate_random_outputs` belongs to `wallet.py`, not a specific functional test. It seems more broadly useful to allow the user to create unconfirmed outputs if they like (and with the amount of tests left to convert I suspect it will come up at some point).